From 7339c393f43bf2fa08edae46ab8f3471924a2659 Mon Sep 17 00:00:00 2001
From: Bakkeby <bakkeby@gmail.com>
Date: Wed, 26 Jun 2024 22:48:40 +0200
Subject: [PATCH] toggletag - anytag patch

This is a variant of the toggletag patch.

Keyboard shortcuts to view a given tag will toggle to the previous layout
if the given tag is already active or if the tag was previously selected.
---
 dwm.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/dwm.c b/dwm.c
index a96f33c..c7dcc98 100644
--- a/dwm.c
+++ b/dwm.c
@@ -2041,8 +2041,11 @@ updatewmhints(Client *c)
 void
 view(const Arg *arg)
 {
-	if ((arg->ui & TAGMASK) == selmon->tagset[selmon->seltags])
+	if ((arg->ui & TAGMASK) == selmon->tagset[selmon->seltags]
+		|| ((arg->ui & TAGMASK) & selmon->tagset[selmon->seltags^1])) {
+		view(&((Arg) { .ui = 0 }));
 		return;
+	}
 	selmon->seltags ^= 1; /* toggle sel tagset */
 	if (arg->ui & TAGMASK)
 		selmon->tagset[selmon->seltags] = arg->ui & TAGMASK;
-- 
2.45.2

